modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1231,68 +1231,93 @@ def _compute_map_reduce_metadata(self, axis, new_parts):
         )
         return result
 
-    def fold_reduce(self, axis, func):
+    def reduce(
+        self,
+        axis: Union[int, Axis],
+        function: Callable,
+        dtypes: Optional[str] = None,
+    ) -> "PandasDataframe":
         """
-        Apply function that reduces Frame Manager to series but requires knowledge of full axis.
+        Perform a user-defined per-column aggregation, where each column reduces down to a single value. Requires knowledge of the full axis for the reduction.
 
         Parameters
         ----------
-        axis : {0, 1}
-            The axis to apply the function to (0 - index, 1 - columns).
-        func : callable
-            The function to reduce the Manager by. This function takes in a Manager.
+        axis : int or modin.pandas.Axis
+            The axis to perform the reduce over.
+        function : callable(row|col) -> single value
+            The reduce function to apply to each column.
+        dtypes : str, optional
+            The data types for the result. This is an optimization
+            because there are functions that always result in a particular data
+            type, and this allows us to avoid (re)computing it.
 
         Returns
         -------
         PandasDataframe
             Modin series (1xN frame) containing the reduced data.
+
+        Notes
+        -----
+        The user-defined function must reduce to a single value.
         """
-        func = self._build_mapreduce_func(axis, func)
+        axis = Axis(axis)
```

Comment:
does this work if axis is already of type Axis?

ditto for all axis parameters that are Union[int, Axis]

Reply:
Yup! I tested it before committing and it acts as an identity if the argument is of type `Axis`!

Reply:
sounds good!